lib/rubocop/cop/style/return_nil_in_predicate_method_definition.rb

```diff
@@ -77,6 +77,8 @@ def on_def(node)
         private
 
         def nil_node_at_the_end_of_method_body(body)
+          return if !body.begin_type? && !body.nil_type?
```

Comment:
It seems to me the logic in this method became a bit too convoluted. I wonder if there isn't a simpler way to get the final node of method that doesn't use `body.children.last` and would not require to the additional checks. I'm guessing you can also just check that the parent of the last `nil` is either `begin` or `def`.

Reply:
I haven't come up with a good logic yet 💦, but for now, I've simplified the conditions a bit.

Reply:
Yeah, it reads better this way. I guess it will do for now.